charon/src/ast/values_utils.rs

```diff
@@ -26,6 +28,32 @@ macro_rules! from_le_bytes {
     }
 }
 
+impl Literal {
+    pub fn to_constant(self, span: Span, ctx: &TransformCtx) -> Result<ConstantExpr, Error> {
+        match self {
+            Literal::Scalar(scal) => Ok(scal.to_constant()),
+            Literal::Float(floatval) => Ok(ConstantExpr {
+                value: RawConstantExpr::Literal(Literal::Float(floatval.clone())),
```

Comment:
Why are you reconstructing the original value here?
```suggestion
                value: RawConstantExpr::Literal(self.clone()),
```
but then you don't even need a `match`. Please remove the match entirely